lib/katalyst/basic/auth/config.rb

```diff
@@ -6,21 +6,61 @@ module Katalyst
   module Basic
     module Auth
       class Config
+        DEFAULT_USERNAME = "katalyst"
+        ROOT_PATH        = "/"
+
         class << self
-          def enabled?
-            enable = ENV.fetch("KATALYST_BASIC_AUTH_ENABLED", enabled_rails_env?)
-            [true, "yes", "true"].include?(enable)
+          include Enumerable
+
+          # @param path [String] Request path
+          # @return [Config] The config for the given path
+          def for_path(path)
+            path ||= ROOT_PATH
+            all.sort_by(&:path)
+               .reverse
+               .detect { |i| path.match(/^#{i.path}/) } || global
+          end
+
+          # @return [Config] The global configuration
+          def global
+            all[0]
+          end
+
+          def add(path:, username: nil, password: nil, enabled: nil)
+            config = Config.new(path: path, username: username, password: password, enabled: enabled)
```

Comment:
(opt) we're at the class level, so you shouldn't need to specify the class:
```
config = new(...)
```

Reply:
good pick up, fixed